pwndbg/lib/cache.py

```diff
@@ -0,0 +1,140 @@
+"""
+Caches return values until some event in the inferior happens,
+e.g. execution stops because of a SIGINT or breakpoint, or a
+new library/objfile are loaded, etc.
+"""
+
+from functools import wraps
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Tuple
+
+debug = 0
```

Comment:
Are we planning to use this?

Reply:
I was thinking of introducing an `if debug: <another impl that print lots of info e.g. when a cache hit/miss happens>`.

This can also be achieved with a hack like: `cache = {} if not debug else DebugDictWhichPrintStuff()`